Source/DafnyCore/Backends/CSharp/CsharpBackend.cs

```diff
@@ -39,123 +35,107 @@ public class CsharpBackend : ExecutableBackend {
     string callToMain /*?*/, string targetFilename /*?*/, ReadOnlyCollection<string> otherFileNames,
     bool runAfterCompile, TextWriter outputWriter) {
 
-    // .NET Core does not allow C# compilation on all platforms using System.CodeDom. You need to use Roslyn libraries. Context: https://github.com/dotnet/runtime/issues/18768
-    var compilation = CSharpCompilation.Create(Path.GetFileNameWithoutExtension(dafnyProgramName))
-      .WithOptions(new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary))
-      .AddReferences(
-        MetadataReference.CreateFromFile(typeof(object).GetTypeInfo().Assembly.Location),
-        MetadataReference.CreateFromFile(Assembly.Load("mscorlib").Location));
+    var outputDir = targetFilename == null ? Directory.GetCurrentDirectory() : Path.GetDirectoryName(Path.GetFullPath(targetFilename));
+    var fileNames = Path.GetFileNameWithoutExtension(Path.GetFileName(dafnyProgramName));
+    var sourcePath = Path.Join(outputDir, fileNames + ".cs");
+    var csprojPath = Path.Join(outputDir, fileNames + ".csproj");
```

Comment:
I think switching to compiling externally and using a project file is a good change. FYI the other backends have had to move in this direction more recently as well: Go now emits a `go.mod` file for `dafny run` and `dafny build`, and Rust similarly a `Cargo.toml` file. I also recently tweaked `smithy-dafny` to support a mode where it emits project files as well, and that behavior is a bit different since it has more packaging-related information to fill in.

For now I'm glad this is only an internal detail that doesn't affect the behavior of `dafny translate`. But I think we need to quickly figure out the long-term strategy there, whether we want to move Dafny itself towards being able to emit fully-formed ready-to-build-and-ship target language projects.